consensus/src/experimental/buffer_item.rs

```diff
@@ -72,31 +86,40 @@ impl BufferItem {
         ordered_proof: LedgerInfoWithSignatures,
         callback: StateComputerCommitCallBackType,
     ) -> Self {
-        Self::Ordered(Box::new(OrderedBufferItem {
-            pending_votes: BTreeMap::<AccountAddress, Ed25519Signature>::new(),
+        Self::Ordered(Box::new(OrderedItem {
+            unverified_signatures: BTreeMap::new(),
             callback,
             ordered_blocks,
             ordered_proof,
         }))
     }
 
     // pipeline functions
-    pub fn advance_to_executed(self, executed_blocks: Vec<ExecutedBlock>) -> Self {
+    pub fn advance_to_executed(
+        self,
+        executed_blocks: Vec<ExecutedBlock>,
+        validator: &ValidatorVerifier,
+    ) -> Self {
         match self {
             Self::Ordered(ordered_item) => {
-                let OrderedBufferItem {
+                let OrderedItem {
                     ordered_blocks,
-                    pending_votes,
+                    unverified_signatures,
                     callback,
                     ordered_proof,
                 } = *ordered_item;
                 for (b1, b2) in zip_eq(ordered_blocks.iter(), executed_blocks.iter()) {
                     assert_eq!(b1.id(), b2.id());
                 }
                 let commit_info = executed_blocks.last().unwrap().block_info();
-                Self::Executed(Box::new(ExecutedBufferItem {
+                let commit_proof = aggregate_ledger_info(
```

Comment:
good point, added it and changed the root logic a little bit. let me know if it makes sense.

Execution root will never need reset since it
always points to OrderedItem (which can't be popped). Signing root would need reset if it
points to prefix of aggregated items.